packages/unlock-js/src/web3Service.ts

```diff
@@ -917,4 +917,94 @@ export default class Web3Service extends UnlockService {
     const baseTokenURI = tokenURI.substring(0, tokenURI.lastIndexOf('/') + 1)
     return baseTokenURI
   }
+
+  /**
+   * Returns the type of the input address
+   */
+  async getEthAddressType(address: string) {
+    const isValidEns = address.endsWith('.eth')
+    const isValidAddress = ethers.utils.isAddress(address)
+    const ETH_ADDRESS_TYPE = {
+      name: 'name',
+      address: 'address',
+      error: 'error',
+    }
+
+    if (!address) return ETH_ADDRESS_TYPE.error
+
+    if (isValidAddress) {
+      return ETH_ADDRESS_TYPE.address
+    } else if (isValidEns) {
+      return ETH_ADDRESS_TYPE.name
+    } else return ETH_ADDRESS_TYPE.error
+  }
+
+  /**
+   * Returns true if the address is a valid EOA
+   */
+  async isValidEOA(address: string) {
+    const provider = this.providerForNetwork(1)
+
+    try {
+      const code = await provider.getCode(address)
+      const count = await provider.getTransactionCount(address)
+      if (code === '0x' && count > 0) {
+        return true
+      } else {
+        return false
+      }
+    } catch (error) {
+      return false
+    }
+  }
+
+  /**
+   * Returns an object the contains the resolved address or ens
+   * name of the input address with it's type
+   */
+  async resolveName(address: string) {
+    const provider = this.providerForNetwork(1)
+    const isValidEOA = await this.isValidEOA(address)
```

Comment:
Does this work with an ens since you are not resolving ens before passing it?

Reply:
Yeah it works, cause both [`getCode`](https://docs.ethers.org/v5/api/providers/provider/#Provider-getCode) & [`getTransactionCount`](https://docs.ethers.org/v5/api/providers/provider/#Provider-getTransactionCount) method accepts ens name.